crypto/evp/e_chacha20_poly1305.c

```diff
@@ -346,7 +346,7 @@ static int chacha20_poly1305_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg,
     case EVP_CTRL_COPY:
         if (actx) {
             if ((((EVP_CIPHER_CTX *)ptr)->cipher_data =
```

Comment:
I agree; I tried, but I don't like the result,...
You need to introduce a new variable to store the cast result, or copy-paste it ...